elasticsearch/elasticsearch.d.ts

```diff
@@ -1,53 +1,78 @@
         ping(params: PingParams, callback: (err: any, response: any, status: any) => void): void;
-        scroll(params: ScrollParams): IPromise<any>;
+        scroll(params: ScrollParams): PromiseLike<any>;
         scroll(params: ScrollParams, callback: (error: any, response: any) => void): void;
-        search(params: SearchParams): IPromise<any>;
+        search(params: SearchParams): PromiseLike<any>;
         search(params: SearchParams, callback: (error: any, response: any) => void): void;
-        suggest(params: SuggestParams): IPromise<any>;
+        suggest(params: SuggestParams): PromiseLike<any>;
         suggest(params: SuggestParams, callback: (error: any, response: any) => void): void;
-        update(params: UpdateDocumentParams): IPromise<any>;
+        update(params: UpdateDocumentParams): PromiseLike<any>;
         update(params: UpdateDocumentParams, callback: (error: any, response: any) => void): void;
     }
 
     export class Indices {
         delete(params: IndicesDeleteParams, callback: (error: any, response: any, status: any) => void): void;
-        delete(params: IndicesDeleteParams): IPromise<any>;
+        delete(params: IndicesDeleteParams): PromiseLike<any>;
         create(params: IndicesCreateParams, callback: (error: any, response: any, status: any) => void): void;
-        create(params: IndicesCreateParams): IPromise<any>;
+        create(params: IndicesCreateParams): PromiseLike<any>;
         exists(params: IndicesIndexExitsParams, callback: (error: any, response: any, status: any) => void): void;
-        exists(params: IndicesIndexExitsParams): IPromise<any>;
+        exists(params: IndicesIndexExitsParams): PromiseLike<any>;
         get(params: IndicesGetParams, callback: (error: any, response: any, status: any) => void): void;
-        get(params: IndicesGetParams): IPromise<any>;
+        get(params: IndicesGetParams): PromiseLike<any>;
         getAlias(params: IndicesGetAliasParams, callback: (error: any, response: any, status: any) => void): void;
-        getAlias(params: IndicesGetAliasParams): IPromise<any>;
+        getAlias(params: IndicesGetAliasParams): PromiseLike<any>;
         putAlias(params: IndicesPutAliasParams, callback: (error: any, response: any, status: any) => void): void;
-        putAlias(params: IndicesPutAliasParams): IPromise<any>;
+        putAlias(params: IndicesPutAliasParams): PromiseLike<any>;
         putTemplate(params: IndicesPutTemplateParams, callback: (error: any, response: any) => void): void;
-        putTemplate(params: IndicesPutTemplateParams): IPromise<any>;
+        putTemplate(params: IndicesPutTemplateParams): PromiseLike<any>;
         refresh(params: IndicesRefreshParams, callback: (error: any, response: any) => void): void;
-        refresh(params: IndicesRefreshParams): IPromise<any>;
+        refresh(params: IndicesRefreshParams): PromiseLike<any>;
+    }
+
+    export interface ConfigOptions{
+        host?: string | string[] | any;
```

Comment:
adding `any` to a union type reduces is to just `any`. i would remove these and keep them `any`.